fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordStoreState.java

```diff
@@ -31,52 +31,101 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
 /**
- * The <code>RecordStoreState</code> interface handles information
- * that might differ between otherwise identical record stores. In particular,
- * it handles the data that might be different between two different
- * record stores that have the same metadata. As a result, even though
- * the {@link RecordMetaData} class contains an instance of this class,
- * it is not serialized with the rest of the metadata and must be retrieved
- * from the store.
+ * The <code>RecordStoreState</code> interface handles information that might differ between otherwise identical
+ * record stores. In particular, it handles the data that might be different between two different record stores
+ * that have the same meta-data.
+ *
+ * <p>
+ * At the moment, this class tracks two pieces of store-specific meta-data:
```

Comment:
I'm inclined to permute these here and elsewhere. That is, a store state is the header info and index states (in that order).

I probably couldn't defend this preference if pressed to.